submissions/evgeniy24/friends_app/random-color-header.js

```diff
@@ -0,0 +1,36 @@
+let header = document.querySelector('.header-name');
+let lettersArr = header.innerHTML.split('');
+//convert int to hex
+const colToHex = function(c) {
+    //colors are bright enough?
+    let color = (c < 75) ? c + 75 : c
+    let hex = color.toString(16);
+    return hex.length == 1 ? '0' + hex : hex;
+}
+
+// colToHex to concatenate a full 6 digit hex code
+const rgbToHex = function(r,g,b) {
+    return '#' + colToHex(r) + colToHex(g) + colToHex(b);
+} 
+
+const getRandomColor = function() {
+    return rgbToHex (
+        Math.floor(Math.random() * 255),
+        Math.floor(Math.random() * 255),
+        Math.floor(Math.random() * 255)
+    )
+}
+
+Array.prototype.randomColor = function() {
```

Comment:
Eeehm... Why it should be in prototype?

And in separate file?

Reply:
I found this solution on the Internet, I liked it and decided to implement it in my friends app.
To make it easier to work with the code during implementation, it worked in a separate file!
In a new commit, I moved everything to the main file.

Reply:
It's a bad solution :)

First of all, you [monkey patched](https://en.wikipedia.org/wiki/Monkey_patch) the prototype of the array. Check part with risks.

Also, when you do such things, you should check case if a method with your name exists in the prototype, just not to override things. Imagine you have added a method called `filter` and then JS will also ship such a method. You will override existing functionality. Check story about [polyfills](https://developer.mozilla.org/en-US/docs/Glossary/Polyfill)

Aaaand... copy-pasting solutions from internet leads to [SODD](https://dzone.com/articles/stack-overflow-driven-development-sodd-its-really)